bugbug/models/performancebug.py

```diff
@@ -0,0 +1,162 @@
+            bug_features.Keywords(),
+            bug_features.IsCoverityIssue(),
+            bug_features.HasCrashSignature(),
+            bug_features.HasURL(),
+            bug_features.HasW3CURL(),
+            bug_features.HasGithubURL(),
+            bug_features.Whiteboard(),
+            bug_features.BlockedBugsNumber(),
+            bug_features.EverAffected(),
+            bug_features.AffectedThenUnaffected(),
+            bug_features.Product(),
+            bug_features.Component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(
+                        feature_extractors,
+                        cleanup_functions,
+                        rollback=True,
+                        rollback_when=self.rollback,
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = ImblearnPipeline(
+            [
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.0001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", BorderlineSMOTE(random_state=0)),
+                (
+                    "estimator",
+                    xgboost.XGBClassifier(n_jobs=utils.get_physical_cpu_count()),
+                ),
+            ]
+        )
+
+    def rollback(self, change):
+        if change["field_name"] == "whiteboard" and any(
+            f"[{whiteboard_text}" in change["added"].lower()
+            for whiteboard_text in (
+                "fxperf",
+                "fxperfsize",
+                "snappy",
+                "pdfjs-c-performance",
+                "pdfjs-performance",
+                "sp3",
+            )
+        ):
+            return True
+
+        if change["field_name"] == "cf_performance_impact" and change["added"] in (
+            "low",
+            "medium",
+            "high",
+        ):
+            return True
+
+        if change["field_name"] == "keywords" and any(
+            keyword.startswith(keyword_start)
+            for keyword_start in ("perf", "topperf", "main-thread-io")
```

Comment:
I would use `prefix` instead of `keyword_start`:
```suggestion
            for prefix in ("perf", "topperf", "main-thread-io")
```